src/Core/Grid/Query/Filter/DoctrineFilterApplicator.php

```diff
@@ -80,8 +86,67 @@ public function apply(QueryBuilder $qb, SqlFilters $filters, array $filterValues
                         $qb->setParameter($name, sprintf('%s %s', $value['to'], '23:59:59'));
                     }
 
+                    break;
+                case SqlFilters::MIN_MAX:
+                    if (!isset($value['min_field']) && !isset($value['max_field'])) {
+                        break;
+                    }
+
+                    $minFieldSqlCondition = "$sqlField >= :{$filterName}_min";
+                    $maxFieldSqlCondition = "$sqlField <= :{$filterName}_max";
+
+                    switch ($this->computeMinMaxCase($value)) {
+                        case self::CASE_BOTH_FIELDS_EXIST_NOT_EQUAL:
+                            $qb->andWhere("$minFieldSqlCondition AND $maxFieldSqlCondition");
+                            $qb->setParameter("{$filterName}_min", $value['min_field']);
+                            $qb->setParameter("{$filterName}_max", $value['max_field']);
+                            break;
+                        case self::CASE_ONLY_MIN_FIELD_EXISTS:
+                            $qb->andWhere($minFieldSqlCondition);
+                            $qb->setParameter("{$filterName}_min", $value['min_field']);
+                            break;
+                        case self::CASE_ONLY_MAX_FIELD_EXISTS:
+                            $qb->andWhere($maxFieldSqlCondition);
+                            $qb->setParameter("{$filterName}_max", $value['max_field']);
+                            break;
+                        case self::CASE_BOTH_FIELDS_ARE_EQUAL:
```

Comment:
Oh I didn't think of that case!! 😅

But wouldn't it work with the case both fields exist? `field >= min AND field <= max` even if it's less optimal it seems equivalent

Reply:
You are right it would work. I think I copied this part from previous Sarunas work and didn't think about it too much.

I decided to remove `BOTH_FIELDS_ARE_EQUAL` case. As you said it's less optimal but I can't see the impact on performance being noticeable even on the largest shop.